Client/core/CCore.cpp

```diff
@@ -2008,48 +2008,47 @@ void CCore::OnPreFxRender()
 //
 // OnPreHUDRender
 //
-void CCore::OnPreHUDRender()
+void CCore::OnPreHUDRender(PreHUDRenderStage stage)
 {
-    IDirect3DDevice9* pDevice = CGraphics::GetSingleton().GetDevice();
-
-    if (CGraphics::GetSingleton().HasLine3DPostFXQueueItems() || CGraphics::GetSingleton().HasPrimitive3DPostFXQueueItems())
+    if (stage == PreHUDRenderStage::PostColorFilter)
```

Comment:
Maybe just split function? This will remove unnecessary check

Reply:
Do you mean
```
if (stage == PreHUDRenderStage::PostColorFilter)
   OnPostColor();
else
   OnPreHUD();
```
?

Reply:
i mean
```cpp
CCore::OnPostColorFilterRender {}
CCore::OnPreHUDRender {}
```

Reply:
As I've already mentioned in the description (https://github.com/multitheftauto/mtasa-blue/pull/3478#issue-2355528510) this would make the interface noisy. Because I cannot add a new method in the middle of the CCoreInterface.

Reply:
Ah, i see. 
I would prefer to add a function, because it doesn't affects the logic or runtime. 
It seems that creating an additional enum and a check creates more mess than misplaced method does

Reply:
I'm personally also inclined to the dedicated method. So, since there's no other opinion I'll do the thing.